Enhancement/user settings page

NeoVier opened the pull request:
## What issue does this PR close
Closes #796 

## Changes Proposed ( a list of new changes introduced by this PR)
- [x] Create a new `/settings` page, with account settings for the user
- [x] Remove the preferences that are now in `/settings` from `/profile/:user`
- [x] Add a link in the menu (hovering the user's picture) to `/settings`
- [x] Change how the menu (hovering the user's picture) looks
- [x] Figure out what to do regarding this: https://cambiatus.slack.com/archives/CA83HJAAD/p1661873170294309

## How to test ( a list of instructions on how to test this PR)
- Go to the new settings page (from the menu accessible on your avatar on the top right)
- Check that everything looks ok and everything works ok
- Go to your profile, and make sure these preferences aren't there anymore, and the profile still looks ok
- Check the new menu (hovering the user's picture) style

NeoVier commented on src/elm/Page/Settings.elm:
```diff
@@ -0,0 +1,528 @@
+    }
+
+
+viewPdfErrorModal : LoggedIn.Model -> Model -> Html Msg
+viewPdfErrorModal loggedIn model =
+    View.Modal.initWith
+        { closeMsg = ClosedPdfErrorModal
+        , isVisible =
+            case model.pdfDownloadStatus of
+                PdfOk ->
+                    False
+
+                EncryptedPassphraseNotPresent ->
+                    True
+        }
+        -- TODO - I18N
+        |> View.Modal.withHeader "Sorry, we can't find your 12 words"
+        |> View.Modal.withBody
+            [ p [ class "my-3" ]
+                -- TODO - I18N
+                [ text "Please, check if you have your 12 words saved during the registration process and use them for further signing in." ]
+            , p [ class "my-3" ]
+                -- TODO - I18N
+                [ text "If you completely lost your 12 words, please, contact us and provide this private key and we will help you to recover:"
+                ]
+            , case LoggedIn.maybePrivateKey loggedIn of
+                Nothing ->
+                    text ""
+
+                Just pk ->
+                    p [ class "font-bold my-3 text-lg text-center border p-4 rounded-sm bg-gray-100" ]
+                        [ text (Eos.Account.privateKeyToString pk)
+                        ]
+            ]
+        |> View.Modal.toHtml
```
@lucca65 do you think we should still worry about this? Here's the reasoning behind it, from `index.js` (the linked PR was merged in July 2020):

```js
const store = JSON.parse(getItem(USER_KEY))
const pin = arg.data.pin

// `.encryptedPassphrase` property was added in https://github.com/cambiatus/frontend/pull/270 while redesigning
// the Profile page. For the users who were already logged-in before these changes were introduced,
// this property may not exist. This case is handled by passing `isDownloaded: false` to Elm
// for further processing.
if (store.encryptedPassphrase) {
  const decryptedPassphrase = sjcl.decrypt(pin, store.encryptedPassphrase)
  return downloadPdf(store.accountName, decryptedPassphrase)
} else {
  // The case when there's not passphrase stored in user's browser, only the Private Key
  return { isDownloaded: false }
}
```

lucca65 replied:
I don't bro, this was a real concern back then, as we were still adjusting the wallet, such a core part of the app. Its been stable for a while and I don't think we need to worry about sessions that old. It should actually fail for those users. 

The main concern was that users might loose access to their account completely as back then we supported 12 words or private key. Depending on the method they could be locked out their own account. Since then we dropped PK support.

NeoVier commented:
The only thing missing now is to figure out if we should [delete KYC and address or just KYC](https://cambiatus.slack.com/archives/CA83HJAAD/p1661873170294309)

lucca65 reviewed (changes requested):
https://user-images.githubusercontent.com/1082127/187716488-c4e650c3-0d2a-4545-9f19-53a7349ecf3f.mov


Why does this look soooooo good 😭 I'm emotional, finally this is pretty.

Can we make the icons a little smaller, they look too big.


For reference I love the way telegram does:

<img width="143" alt="Captura de Tela 2022-08-31 às 12 23 46" src="https://user-images.githubusercontent.com/1082127/187716746-14e11da5-12ff-4e2c-9ea8-58ae89a6781d.png">

The size of the font and the icon looks about the same

NeoVier commented:
I adjusted the size of the icons to be closer to the font size. However, I don't think there's a straightforward way to change the thickness of the icons with css. It would be great if we had a scale of weights for icons, but I think the thickness is kind of ok for now. Apple's Human Design Guidelines is a great example for icons: https://developer.apple.com/design/human-interface-guidelines/foundations/sf-symbols

I also adjusted the height of each item to be a little smaller. Let me know what you think 😁

lucca65 reviewed (changes requested):
I love it! seems way better! 

About our icons we use custom ones, but maybe we can start using some from an open source library like https://ionic.io/ionicons. This way we can make sure to keep an easier standard without making great changes to our DS. But maybe those are things we need @rafachadud and @juramos-2020 to validate first. 

I've found a little bug, when navigating with tab key, it skips the new option for configurations. goes from profile to language.

NeoVier commented:
> I've found a little bug, when navigating with tab key, it skips the new option for configurations. goes from profile to language.

I can't seem to replicate that 🤔 


https://user-images.githubusercontent.com/39280468/187767349-57c5fb53-aefd-40d0-a197-5a3c75388fdd.mov

lucca65 commented:
safari maybe?? 🤔 

https://user-images.githubusercontent.com/1082127/187782700-d22554e9-bd1d-4b2d-9255-a73f9fc4764f.mov

lucca65 commented:
Shift Tab also has strange behavior
https://user-images.githubusercontent.com/1082127/187783204-bddf521d-2fea-408f-98fc-9bb4729768fd.mov

NeoVier commented:
That video I sent was on safari 🤔

If you click on the link, does it work? If it does, I don't see why tab wouldn't work. Can you try on an anonymous tab just to be sure?

lucca65 commented:
I've made a new video showing both behaviors. Both for regular tabs. Hover on the profile tag, hit tab.. Then hitting shift tab a few times. It gets out of the profile and goes to the search and community selector
https://user-images.githubusercontent.com/1082127/187790235-60882f51-61aa-48dd-a2b8-f7abae38cdca.mov

lucca65 commented:
meh, the new video don't load either, Ill send over slack

NeoVier commented:
So we don't pollute this PR with this, I've created an issue: https://github.com/cambiatus/frontend/issues/825. I don't think we should address this in this PR, as it will be quite a bit of work, not necessarily related to this PR

juramos-2020 commented:
@NeoVier podemos manter o espaçamento que vc está usando nas laterais na área de cima, entre o nome e topo do menu. Acho que é de 16px, correto? 


![Screenshot_2022-09-01-11-55-57-054_com.android.chrome.jpg](https://user-images.githubusercontent.com/67649886/187946498-59790de7-371e-4af0-8122-dbd4c81e83d9.jpg)

NeoVier commented:
@juramos-2020 aumentei o espaçamento de cima (para 12px que é o que tem nos lados), vê se ficou melhor 😁. Agora dá pra ver pelo deploy preview: https://deploy-preview-823--cambiatus.netlify.app/dashboard

juramos-2020 commented:
Ja Posso fazer o teste?

NeoVier commented:
Pode sim

juramos-2020 commented:
@NeoVier quanto que tem de espaçamento em cada opção do menu? Acho que precisamos aumentar um pouco pois uma opção está muito próxima da outra.

NeoVier commented:
Nesse estilo de menu geralmente não tem nenhum espaçamento entre os itens. Pelo menos é o que eu vejo por aí. Alguns exemplos que abri aqui rapidinho: Slack, GitHub, macOS, Figma, Whatsapp, Spotify, Miro. Até o nosso próprio app, na tela de configuração de categorias, tem um menu assim, e também não tem nenhum espaçamento entre os itens.

Podemos adicionar se quiser, só estou citando a maneira como geralmente é feito

juramos-2020 commented:
OI @NeoVier 

a altura de cada opção esta em 48px?
se tiver ok, se não tiver só sugiro fazer essa alteração. 
No design tava com uma altura de 24px, fiz a correção agora. 
A sugestão foi seguindo o [material.io](https://material.io/components/menus#usage) para melhorar a usabilidade, visto que algumas pessoas tem dedos "gordinhos" e faz com que cliquem na opção errada quando estas apresentam muito unidas.

Mas como disse, se ja tiver 48px beleza ta tudo certo.

juramos-2020 commented:
Também poderia alinhar o texto ao toggle

![Screenshot_2022-09-01-14-03-07-074_com.android.chrome.jpg](https://user-images.githubusercontent.com/67649886/187971551-dcc45bc4-107f-4a16-8a84-2f162355564a.jpg)

NeoVier commented:
Feito ✅ 

No desktop, mantive a altura dos itens em 32px (achei que 48px ficou gigante) - o material UI também faz isso no desktop, com a versão Dense do dropdown.

No mobile, aumentei a altura

juramos-2020 commented:
Maravilha @NeoVier 

Já ia pedir para aproximar mais o texto do toggle, para 8px, conforme ta no nosso desing. Mas vi que esse espaçamento grande está em todo o webapp 😅 Talvez possamos resolver isso no futuro em outra issue né?!

NeoVier commented:
Sim, melhor resolver separado 😅. Consegue abrir uma issue @juramos-2020?

juramos-2020 commented:
Beleza consigo sim ☺️

NeoVier commented:
@juramos-2020 em relação a esse PR, ainda falta alguma coisa ou está tudo ok?
